src/main/java/org/owasp/esapi/Encoder.java

```diff
@@ -444,32 +444,73 @@ public interface Encoder {
     /**
      * Encode data for use in LDAP queries. Wildcard (*) characters will be encoded.
      *
+     * This encoder operates according to RFC 4515, Section 3. RFC 4515 says the following character ranges
+     * are valid: 0x01-0x27, 0x2B-0x5B and 0x5D-0x7F. Characters outside the ranges are hex encoded, and they
+     * include 0x00 (NUL), 0x28 (LPAREN), 0x29 (RPAREN), 0x2A (ASTERISK), and 0x5C (ESC). The encoder will also
+     * encode 0x2F (FSLASH), which is required by Microsoft Active Directory.
+     *
+     * NB: At ESAPI 2.5.3, {@code encodeForLDAP} began strict conformance with RFC 4515. Characters above 0x7F
```

Comment:
Suggest changing 'NB:' to 'NOTE:' (or perhaps 'WARNING:' if you think it's really important to mention because it is likely to break someone's code.

Outside of legal notation, NB is not commonly used any longer, at least not like it was 30 years ago or so when I used to see it in comments throughout UNIX code bases. I'm not sure younger developers will know what that NB stands for the Latin _nota bene_, commonly translated as "note well" unless perhaps they took Latin in high school.